Docs/blog/web ops   cms preview authentication enabled by default (#1258)
vtex/faststore: docs/blog/2022-05-02-webops.md

---

> description: CMS previews now require authentication by default
> tags: [webops]
> ---
> 
> # CMS previews now require authentication by default

Reviewer: ```suggestion
description: VTEX Headless CMS previews now require authentication by default
tags: [cms]
---

# VTEX Headless CMS previews are now private by default
```

> Following the next successful push to your store's repository default branch, the CMS preview available at `https://gatsby-preview-vtex-sites--<your-account-here>-store.vtex.io` will require users to authenticate to VTEX.

Reviewer: ```suggestion
Following the next successful push to your store's repository default branch, the VTEX Headless CMS preview available at `https://gatsby-preview-vtex-sites--<your-account-here>-store.vtex.io` will require users to authenticate to VTEX.
```

In order to access the preview, users will need to have access to the `myvtex.com` domain of that specific account. This can be done either through the build environment variable `GATSBY_STORE_ID` in the `vtex.env` file, or the `api.storeId` entry exported by the `store.config.js` configuration file.

## What has changed?

Previously, CMS previews were published to public URLs by default. Now, they are private by default, requiring authentication to VTEX.

## Why did we make these changes?

> It is common for sales promotions and product launches to appear in CMS previews before they are made available to the public. This change aims to prevent that while making the platform more secure.

Reviewer: ```suggestion
CMS previews may contain sales promotions and product launches that are not yet available to the general public. This change aims to prevent unauthorized access to such content while also making the platform more secure.
```

Reviewer: unauthorized, undesired...

## What needs to be done?

> If you rely on third parties (e.g page speed insights) having access to your CMS preview or want your CMS preview to be public for other reasons, you can disable authentication by adding `USE_CMS_PREVIEW_AUTH=false` to your repository's `vtex.env` file.

Reviewer: ```suggestion
If, for any reason, you need your CMS previews to be accessed by the public or third-party solutions, you may disable authentication by adding `USE_CMS_PREVIEW_AUTH=false` to your repository's `vtex.env` file.
```

> 
> If that is not your case, there is no action needed.

Reviewer: ```suggestion
```
